src/WebJobs.Script.WebHost/Diagnostics/LinuxContainerEventGenerator.cs

```diff
@@ -24,6 +28,7 @@ public LinuxContainerEventGenerator(IEnvironment environment, Action<string> wri
                 _consoleEnabled = false;
             }
             _containerName = _environment.GetEnvironmentVariable(EnvironmentSettingNames.ContainerName)?.ToUpperInvariant();
+            _metricsPublisher = new LinuxContainerMetricsPublisher(environment, standbyOptions, LogMetricsPublishEvent, httpClient);
```

Comment:
Instead of passing the LogMetricsPublishEvent func like this, if you instead just have LinuxContainerMetricsPublisher take an ILogger<LinuxContainerMetricsPublisher> ctor param via DI, you can remove this entire line and get the LinuxContainerMetricsPublisher instance via DI. That would be preferable.

Reply:
I tried DI that but it leads to circular dependencies in DI which is why I am doing it this way